apps/prairielearn/src/pages/publicAssessmentQuestions/publicAssessmentQuestions.html.ts

```diff
@@ -0,0 +1,127 @@
+        ${HeadContents({ resLocals })}
+      </head>
+      <body>
+        ${Navbar({ resLocals })}
+        <main id="content" class="container">
+          ${course.sharing_name
+            ? html`
+                <div class="card mb-4">
+                  <div class="card-header bg-primary text-white d-flex align-items-center">
+                    ${assessment.title} ${assessment.number}: Questions
+                  </div>
+                  ${AssessmentQuestionsTable({
+                    questions,
+                    urlPrefix: resLocals.urlPrefix,
+                    course_id: course.id,
+                    course_instance_id,
+                    course_sharing_name: course.sharing_name,
+                  })}
+                </div>
+              `
+            : html`<p>
+                This course doesn't have a sharing name. If you are an Owner of this course, please
+                choose a sharing name on the
+                <a
+                  href="${resLocals.plainUrlPrefix}/course/${resLocals.course
+                    .id}/course_admin/sharing"
+                  >course sharing settings page</a
+                >.
+              </p>`}
+        </main>
+      </body>
+    </html>
+  `.toString();
+}
+
+function AssessmentQuestionsTable({
+  questions,
+  urlPrefix,
+  course_id,
+  course_instance_id,
+  course_sharing_name,
+}: {
+  questions: AssessmentQuestionRow[];
+  urlPrefix: string;
+  course_id: string;
+  course_instance_id: string;
+  course_sharing_name: string;
+}) {
+  const nTableCols = 4;
+  return html`
+    <div class="table-responsive">
+      <table class="table table-sm table-hover">
+        <thead>
+          <tr>
+            <th><span class="visually-hidden">Name</span></th>
+            <th>QID</th>
+            <th>Topic</th>
+            <th>Tags</th>
+            <th>Other Assessments</th>
+          </tr>
+        </thead>
+        <tbody>
+          ${questions.map((question) => {
+            return html`
+              ${AssessmentQuestionHeaders(question, nTableCols)}
+              <tr>
+                <td>
+                  <a
+                    href="${urlPrefix}/public/course/${course_id}/question/${question.question_id}/preview"
+                  >
+                    ${AssessmentQuestionNumber(question)}${question.title}
+                  </a>
+                </td>
+                <td>${`@${course_sharing_name}/${question.display_name}`}</td>
+                <td>${TopicBadge(question.topic)}</td>
+                <td>${TagBadgeList(question.tags)}</td>
+                <td>
+                  ${question.other_assessments
+                    ? question.other_assessments.map((assessment) => {
+                        return html`${AssessmentBadge({
+                          assessment,
+                          plainUrlPrefix: urlPrefix,
+                          course_instance_id,
+                          publicURL: true,
+                        })}`;
```

Comment:
There's no need to wrap a component in another `html`:

```suggestion
                        return AssessmentBadge({
                          assessment,
                          plainUrlPrefix: urlPrefix,
                          course_instance_id,
                          publicURL: true,
                        });
```

Reply:
done in 3a35b9749e6b83547eb14988f73709fa2d9075a5